python/python/tests/utils/utils.py

```diff
@@ -359,3 +361,40 @@ def check_function_stats_response(
         b"LUA": {b"libraries_count": lib_count, b"functions_count": function_count}
     }
     assert expected == response.get(b"engines")
+
+
+def ft_search_deep_compare_result(
+    result: List[Union[int, Mapping[TEncodable, Mapping[TEncodable, TEncodable]]]],
+    json_key1: str,
+    json_key2: str,
+    json_value1: dict,
+    json_value2: dict,
+):
+    """
+    Deep compare the keys and values in FT.SEARCH result array.
+
+    Args:
+        result (List[Union[int, Mapping[TEncodable, Mapping[TEncodable, TEncodable]]]]):
+        json_key1 (str): The first key in search result.
+        json_key2 (str): The second key in the search result.
+        json_value1 (dict): The fields map for first key in the search result.
+        json_value2 (dict): The fields map for second key in the search result.
+    """
+    assert len(result) == 2
+    assert result[0] == 2
+    searchResultMap: Mapping[TEncodable, Mapping[TEncodable, TEncodable]] = cast(
+        Mapping[TEncodable, Mapping[TEncodable, TEncodable]], result[1]
+    )
+    fieldName1 = "a"
+    fieldName2 = "b"
```

Comment:
Move this outside of this func. Comparing method should compare only. The test defines expected result and gets actual from the server.

Reply:
Updated.